examples_notebooks/community_contrib/couchbase/GraphRAG_with_Couchbase.ipynb

```diff
@@ -0,0 +1,557 @@
+    "graphrag.query and vector_stores: These are custom libraries that handle entity extraction, searching, and vector storage."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {
+    "id": "5tIHss5Rglye"
+   },
+   "outputs": [],
+   "source": [
+    "import asyncio\n",
+    "import logging\n",
+    "import os\n",
+    "import traceback\n",
+    "from typing import Any, Callable, Dict, List, Union\n",
+    "\n",
+    "import pandas as pd\n",
+    "import tiktoken\n",
+    "\n",
+    "from graphrag.query.context_builder.entity_extraction import EntityVectorStoreKey\n",
+    "from graphrag.query.indexer_adapters import (\n",
+    "    read_indexer_covariates,\n",
+    "    read_indexer_entities,\n",
+    "    read_indexer_relationships,\n",
+    "    read_indexer_reports,\n",
+    "    read_indexer_text_units,\n",
+    ")\n",
+    "from graphrag.query.input.loaders.dfs import store_entity_semantic_embeddings\n",
+    "from graphrag.query.llm.oai.chat_openai import ChatOpenAI\n",
+    "from graphrag.query.llm.oai.embedding import OpenAIEmbedding\n",
+    "from graphrag.query.llm.oai.typing import OpenaiApiType\n",
+    "from graphrag.query.structured_search.local_search.mixed_context import (\n",
+    "    LocalSearchMixedContext,\n",
+    ")\n",
+    "from graphrag.query.structured_search.local_search.search import LocalSearch\n",
+    "from graphrag.vector_stores.couchbasedb import CouchbaseVectorStore"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "2efKWBqpgcw-"
+   },
+   "source": [
+    "# Configuring Environment Variables\n",
+    "Here, we configure various environment variables that define paths, API keys, and connection strings. These values are essential for connecting to Couchbase and OpenAI, loading data, and defining other constants.\n",
+    "\n",
+    "INPUT_DIR: This specifies where to find the data files.\n",
+    "COUCHBASE_CONNECTION_STRING: Connection details for Couchbase.\n",
+    "OPENAI_API_KEY: Your OpenAI API key, required for interacting with their models.\n",
+    "LLM_MODEL: Specifies which OpenAI model to use (e.g., GPT-4).\n",
+    "EMBEDDING_MODEL: Defines the model used to generate embeddings."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {
+    "id": "Cz1PfM7zgc39"
+   },
+   "outputs": [],
+   "source": [
+    "INPUT_DIR = os.getenv(\"INPUT_DIR\")\n",
+    "COUCHBASE_CONNECTION_STRING = os.getenv(\"COUCHBASE_CONNECTION_STRING\", \"couchbase://localhost\")\n",
+    "COUCHBASE_USERNAME = os.getenv(\"COUCHBASE_USERNAME\", \"Administrator\")\n",
+    "COUCHBASE_PASSWORD = os.getenv(\"COUCHBASE_PASSWORD\", \"password\")\n",
+    "COUCHBASE_BUCKET_NAME = os.getenv(\"COUCHBASE_BUCKET_NAME\", \"graphrag-demo\")\n",
+    "COUCHBASE_SCOPE_NAME = os.getenv(\"COUCHBASE_SCOPE_NAME\", \"shared\")\n",
+    "COUCHBASE_VECTOR_INDEX_NAME = os.getenv(\"COUCHBASE_VECTOR_INDEX_NAME\", \"grapghrag_index\")\n",
```

Comment:
Can we use similar env vars as we use in other demos to have uniformity lik. CB_CONN_STR etc. Not necessary though

Reply:
Have kept it for better readability for new users